src/Compilers/Core/Portable/Operations/Expressions.cs

```diff
@@ -1,405 +1,116 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System.Collections.Immutable;
-using System.Diagnostics;
 using System.Linq;
 
 namespace Microsoft.CodeAnalysis.Semantics
 {
-    internal sealed class ConditionalChoice : IConditionalChoiceExpression
+    internal sealed partial class ConditionalChoiceExpression : IConditionalChoiceExpression
```

Comment:
I believe eventually, all code here should be removed and moved to Factory since logic here doesn't actually belong to Operation type itself but the factory that determine what to give in as each property values.